sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -485,4 +490,67 @@ public static String getEncryptionDataKey(Map<String, String> metadata) {
         }
         return null;
     }
+
+    public static <T, U> ResponseBase<T, U> sendRequest(Callable<ResponseBase<T, U>> operation, Duration timeout) {
+        try {
+            Future<ResponseBase<T, U>> future = THREAD_POOL.submit(operation);
+            if (timeout == null) {
+                return getResultWithTimeout(future, 0, RuntimeException.class);
+            }
+            return getResultWithTimeout(future, timeout.toMillis(), RuntimeException.class);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    public static <T, U> ResponseBase<T, U> sendRequest(Callable<ResponseBase<T, U>> operation, Duration timeout, Class<? extends RuntimeException> exceptionType) {
+        try {
+            Future<ResponseBase<T, U>> future = THREAD_POOL.submit(operation);
+            if (timeout == null) {
+                return getResultWithTimeout(future, 0, exceptionType);
+            }
+            return getResultWithTimeout(future, timeout.toMillis(), exceptionType);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
```

Comment:
Should have a separate catch block for `InterruptedException` and `TimeoutException` as those shouldn't have a causal exception that we're looking for, only `ExecutionException` should.